src/_pyinstaller_hooks_contrib/hooks/stdhooks/hook-pypemicro.py

```diff
@@ -0,0 +1,35 @@
+# ------------------------------------------------------------------
+# Copyright (c) 2022 PyInstaller Development Team.
+#
+# This file is distributed under the terms of the GNU General Public
+# License (version 2.0 or later).
+#
+# The full license is available in LICENSE.GPL.txt, distributed with
+# this software.
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+# ------------------------------------------------------------------
+
+# Hook for the pypemicro module: https://github.com/nxpmicro/pypemicro
+
+import glob
+import os
+from PyInstaller import isolated
+from PyInstaller.utils.hooks import remove_prefix, get_package_paths
+
+
+@isolated.decorate
```

Comment:
Since `isolated` is available only in PyInstaller 5, we need to decide what to do about earlier versions. Either we'll need to tie the next release of contrib hooks to PyInstaller 5.0 and later, or this hook will need to check for PyInstaller version and become no-op (or raise error) for older versions (assuming we won't be implementing alternative codepath with old-style subprocesses - which I'd prefer to avoid). @bwoodsend, what do you think?

Reply:
Uhm, yeah I was just going to wing it and leave it unchecked. This repo doesn't enforce any minimum PyInstaller version (it doesn't even have PyInstaller as a dependency) and TBH I'm not convinced that it should force users onto newer PyInstaller versions because some random hook that they're likely not using requires some new feature. I wouldn't mind adding a little check at the top of the hook with a meaningful error message.

Reply:
well, do you mean throw an exception for pyinstaller < 5.0 or just ERROR message using logger?

Reply:
Compatibility-wise, I think the best approach would be to put the contents of the hook under the PyInstaller version check; if pyinstaller < 5.0, display a warning and do nothing else.

Reply:
Or even put the `isolated` code part under the version check and return empty library list in the else branch, along with the warning. This way, interested parties can add a fallback using old `exec_statement`...

Reply:
@rokm, glob removed, Mac OS handled explicitly. check + warning for pyinstaller < 5.0 implemented